### ISSUE TITLE
Processing failed: File is locked for reading/writing

### ISSUE THREAD

**Issue by wigwam12** [SickRage/SickRage#2605]
### Before submitting your issue:

Enable debug logging in SickRage settings, reproduce the error (be sure to disable after the bug is fixed)

Branch/Commit: Master
OS: Xpenology DSM 5.2-5592 Update 4
What you did: Post-processing tv show failed for both Sab via nzbtomedia and in Sickrage
What happened: SickBeard: Failed to post-process - Returned log from SickBeard was not as expected
What you expected: normal post-processing
I was having with nzbtomedia post-processing failures after an upgrade of Sab a few mnths back so all post-processing was bign done by Sickrage with no problem.
I had some advice on what the script issue might be and after doing the fix I turned off sickrage PP but found the shows were returning the post-processing error in Sab as below; turned it back on in Sickrage and it gave the same errors.
I had made no changes to any file permissions etc so I have no idea why it now gives me this error.
I see a few people seem to be seeing the same thing but for different reasons?

Help appreciated pleas.
Thanks
Logs:
```
PASTE LOGS HERE
```[03:42:51] [INFO]::MAIN: Loading config from [/volume1/@appstore/sickbeard-custom/var/SickBeard/contrib/nzbToMedia/autoProcessMedia.cfg]
[03:42:51] [INFO]::MAIN: Checking database structure...
[03:42:51] [DEBUG]::MAIN: Checking Initial Schema database upgrade
[03:42:51] [DEBUG]::MAIN: InitialSchema upgrade not required
[03:42:51] [INFO]::MAIN: Checking if source needs an update
[03:42:51] [DEBUG]::MAIN: cur_commit = None % (newest_commit)= 26957918e562918c653292d33e2c11206ce6ccb9, num_commits_behind = 0
[03:42:51] [ERROR]::MAIN: Unknown current version number, don't know if we should update or not
[03:42:51] [INFO]::MAIN: nzbToMedia Version:10.14 Branch:master (Linux 3.10.35)
[03:42:52] [WARNING]::MAIN: Failed to locate ffprobe. Video corruption detection disabled!
[03:42:52] [WARNING]::MAIN: Install ffmpeg with x264 support to enable this feature  ...
[03:42:52] [DEBUG]::MAIN: Connecting to transmission: http://192.168.0.20:9091
[03:42:52] [INFO]::MAIN: #########################################################
[03:42:52] [INFO]::MAIN: ## ..::[nzbToMedia.pyc]::.. ##
[03:42:52] [INFO]::MAIN: #########################################################
[03:42:52] [DEBUG]::MAIN: Options passed into nzbToMedia: ['/usr/local/sickbeard-custom/var/SickBeard/contrib/nzbToMedia/nzbToSickBeard.py', '/volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION', 'Lucifer.S02E07.1080p.HDTV.X264-DIMENSION.nzb', 'Lucifer.S02E07.1080p.HDTV.X264-DIMENSION', '', 'tv', 'alt.binaries.teevee', '0', 'h', 't', 't', 'p', 's', ':', '/', '/', 'n', 'z', 'b', '.', 'i', 's', '/', 'f', 'a', 'i', 'l', 'e', 'd', '/', '?', 'g', 'u', 'i', 'd', '=', '8', 'f', 'b', 'a', '6', '5', 'b', '3', 'f', '2', '0', '8', '8', '7', 'b', '1', 'a', '1', '9', '8', '0', '3', 'd', '2', '7', '8', '5', '0', 'b', '8', 'd', 'f', '8', '6', 'f', '7', '2', 'f', 'd', '7', '&', 'u', 's', 'e', 'r', 'i', 'd', '=', '4', '8', '1', '7', '0', '&', 'r', 's', 's', 't', 'o', 'k', 'e', 'n', '=', 'c', '8', 'b', '0', '0', '8', '2', 'd', 'b', 'e', 'd', '2', '0', '6', '1', '4', 'd', '0', 'f', '7', 'b', '7', '2', '6', '6', 'a', '3', '3', '4', '2', '4', '6']
[03:42:52] [INFO]::MAIN: Script triggered from SABnzbd 0.7.17+
[03:42:52] [DEBUG]::MAIN: Searching for nzoid from SAbnzbd ...
[03:42:52] [DEBUG]::MAIN: Found nzoid: SABnzbd_nzo_s9Dn5U
[03:42:52] [DEBUG]::MAIN: Adding NZB download info for directory /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION to database
[03:42:52] [INFO]::MAIN: Auto-detected SECTION:SickBeard
[03:42:52] [DEBUG]::MAIN: Checking for archives to extract in directory: /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION
/usr/syno/bin/7z
[03:42:52] [DEBUG]::MAIN: EXTRACTOR: Unknown file type: .r02
/usr/syno/bin/7z
[03:42:52] [DEBUG]::MAIN: EXTRACTOR: Unknown file type: .r06
/usr/syno/bin/7z
[03:42:52] [DEBUG]::MAIN: EXTRACTOR: Unknown file type: .r11
/usr/syno/bin/7z
[03:42:52] [DEBUG]::MAIN: EXTRACTOR: Unknown file type: .r03
/usr/syno/bin/7z
[03:42:52] [DEBUG]::MAIN: EXTRACTOR: Unknown file type: .r01
/usr/syno/bin/7z
[03:42:52] [DEBUG]::MAIN: EXTRACTOR: Unknown file type: .r12
/usr/syno/bin/7z
[03:42:52] [DEBUG]::MAIN: EXTRACTOR: Unknown file type: .r05
/usr/syno/bin/7z
[03:42:52] [DEBUG]::MAIN: EXTRACTOR: Unknown file type: .r04
/usr/syno/bin/7z
[03:42:52] [DEBUG]::MAIN: EXTRACTOR: Unknown file type: .r07
/usr/syno/bin/7z
[03:42:52] [INFO]::MAIN: Extracting /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION/morning.star.207.1080-dimension.rar to /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION
[03:42:52] [DEBUG]::MAIN: Extracting ['unrar', 'x', '-o+', '-y'] /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION/morning.star.207.1080-dimension.rar /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION
[03:43:32] [INFO]::MAIN: EXTRACTOR: Extraction was successful for /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION/morning.star.207.1080-dimension.rar to /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION
[03:43:35] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r02 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:36] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r06 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:37] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r11 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:38] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r03 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:39] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r01 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:40] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r12 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:41] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r05 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:42] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r04 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:43] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r07 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:44] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.rar from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:45] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r21 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:46] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r22 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:47] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r10 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:48] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r23 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:49] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r14 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:50] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r13 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:51] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r25 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:52] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r26 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:54] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r15 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:55] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r16 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:56] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r00 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:57] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r28 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:58] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r17 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:43:59] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r08 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:00] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r30 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:01] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r19 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:02] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r20 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:03] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r32 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:04] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r34 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:05] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r24 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:06] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r35 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:07] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r38 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:08] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r36 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:09] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r39 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:10] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r27 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:11] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r29 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:12] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r42 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:13] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r31 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:14] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r45 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:15] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r18 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:16] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r37 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:17] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r09 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:18] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r40 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:19] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r41 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:20] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r43 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:21] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r44 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:22] [INFO]::MAIN: Removing extracted archive morning.star.207.1080-dimension.r33 from folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION ...
[03:44:23] [INFO]::MAIN: Calling SickBeard:tv to post-process:Lucifer.S02E07.1080p.HDTV.X264-DIMENSION.nzb
[03:44:23] [INFO]::MAIN: SickBeard:tv fork set to sickrage
[03:44:23] [INFO]::MAIN: FLATTEN: Flattening directory: /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION
[03:44:23] [DEBUG]::MAIN: Checking for empty folders in:/volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION
[03:44:23] [POSTPROCESS]::SICKBEARD: SUCCESS: The download succeeded, sending a post-process request
[03:44:23] [DEBUG]::SICKBEARD: Opening URL: http://localhost:8899/home/postprocess/processEpisode with params: {'nzbName': 'Lucifer.S02E07.1080p.HDTV.X264-DIMENSION.nzb', 'quiet': 1, 'failed': '0', 'dir': '/volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION'}
[03:44:23] [POSTPROCESS]::SICKBEARD: Processing folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION
[03:44:23] [POSTPROCESS]::SICKBEARD: PostProcessing Path: /volume1/Downloads/complete/tv
[03:44:23] [POSTPROCESS]::SICKBEARD: PostProcessing Dirs: [u'Lucifer.S02E07.1080p.HDTV.X264-DIMENSION']
[03:44:23] [POSTPROCESS]::SICKBEARD: PostProcessing Files: []
[03:44:23] [POSTPROCESS]::SICKBEARD: PostProcessing VideoFiles: []
[03:44:23] [POSTPROCESS]::SICKBEARD: PostProcessing RarContent: []
[03:44:23] [POSTPROCESS]::SICKBEARD: PostProcessing VideoInRar: []
[03:44:23] [POSTPROCESS]::SICKBEARD: Processing folder Lucifer.S02E07.1080p.HDTV.X264-DIMENSION
[03:44:23] [POSTPROCESS]::SICKBEARD: Processing folder /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION
[03:44:23] [POSTPROCESS]::SICKBEARD: Found unwanted files: ['lucifer.207.1080-dimension.nfo', 'morning.star.207.1080-dimension.sfv', 'morning.star.207.1080-dimension.srr', 'morning.star.207.1080-dimension.sample.mkv']
[03:44:23] [POSTPROCESS]::SICKBEARD: Processing /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION.mkv (Lucifer.S02E07.1080p.HDTV.X264-DIMENSION.nzb)
[03:44:23] [POSTPROCESS]::SICKBEARD: Retrieving episode object for 2x7
[03:44:23] [POSTPROCESS]::SICKBEARD: Snatch history had a quality in it, using that: 1080p HDTV
[03:44:23] [POSTPROCESS]::SICKBEARD: SR snatched this episode and it is not processed before
[03:44:23] [POSTPROCESS]::SICKBEARD: Is ep a priority download: True
[03:44:23] [POSTPROCESS]::SICKBEARD: There is no existing file so there's no worries about replacing it
[03:44:23] [POSTPROCESS]::SICKBEARD: This download is marked a priority download so I'm going to replace an existing file if I find one
[03:44:23] [POSTPROCESS]::SICKBEARD: Unable to determine needed filespace as the source file is locked for access
[03:44:23] [POSTPROCESS]::SICKBEARD: Found release name Lucifer.S02E07.1080p.HDTV.X264-DIMENSION
[03:44:23] [POSTPROCESS]::SICKBEARD: Destination folder for this episode: /volume1/video/tv/Lucifer/Season 02
[03:44:23] [POSTPROCESS]::SICKBEARD: Processing failed for /volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION.mkv: File is locked for reading/writing
[03:44:23] [POSTPROCESS]::SICKBEARD: Problem(s) during processing, failed the following files/folders:  
[03:44:23] [POSTPROCESS]::SICKBEARD: [/volume1/Downloads/complete/tv/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION/Lucifer.S02E07.1080p.HDTV.X264-DIMENSION.mkv : Processing failed: File is locked for reading/writing]
[03:44:23] [ERROR]::MAIN: A problem was reported in the /usr/local/sickbeard-custom/var/SickBeard/contrib/nzbToMedia/nzbToSickBeard.py script.
SickBeard: Failed to post-process - Returned log from SickBeard was not as expected

**Comment by SickRage2** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [ ] Post debug logs, either inline (for smaller logs) or using gist

Please make sure you also read [how to create an issue](https://github.com/SickRage/sickrage-issues/blob/master/README.md) and followed all of the steps. 

The title should describe your issue. Having "SR not working" or "I get this bug" for 100 issues, isn't really helpful. We will close issues if there isn't enough information.

Sometimes the devs may seem like grunts and respond with short answers. This isn't (always) because the dev hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let us know, and this bot may get updated to automatically answer you.

Thanks!

**Comment by neoatomic** [comment]
It clearly indicates that the files are in "use" or that permissions have changed. Maybe that happens during a update.? 
Or two PostProsessing sessions need to be running at the same time. That can happen if you have set the PP setting very fast. So that a new PP is started before the old one is completed.

**Comment by wigwam12** [comment]
HI
Actually I did the reverse - I increased the PP time from 10 to 360 so the machine wasn't checking so frequently.
I also disabled PP in Sickrage anyway to stop the potential conflict.
I did make sure the permissions to the folders were 777 again but despite all this I have this error.

The only other thing I wonder is that the scripts are in the sickbeard-custom folder. Sickrage has all the permissions and the script does refer to sickrage; would this make a difference?
